src/Orleans.Core/Messaging/BufferPool.cs

```diff
@@ -1,80 +1,45 @@
 using System;
-using System.Collections.Concurrent;
+using System.Buffers;
 using System.Collections.Generic;
-using System.Threading;
+using System.Runtime.CompilerServices;
 using Orleans.Configuration;
 namespace Orleans.Runtime
 {
     internal class BufferPool
     {
+        private const int MaximumBufferSize = int.MaxValue;
         private readonly int byteBufferSize;
-        private readonly int maxBuffersCount;
-        private readonly bool limitBuffersCount;
-        private readonly ConcurrentBag<byte[]> buffers;
-        private readonly CounterStatistic allocatedBufferCounter;
         private readonly CounterStatistic checkedOutBufferCounter;
         private readonly CounterStatistic checkedInBufferCounter;
-        private readonly CounterStatistic droppedBufferCounter;
-        private readonly CounterStatistic droppedTooLargeBufferCounter;
-
-        private int currentBufferCount;
 
         public static BufferPool GlobalPool;
-
         public int Size
         {
             get { return byteBufferSize; }
         }
 
-        public int Count
-        {
-            get { return buffers.Count; }
-        }
-
-        public string Name
-        {
-            get;
-            private set;
-        }
-
         internal static void InitGlobalBufferPool(MessagingOptions messagingOptions)
         {
-            GlobalPool = new BufferPool(messagingOptions.BufferPoolBufferSize, messagingOptions.BufferPoolMaxSize, messagingOptions.BufferPoolPreallocationSize, "Global");
+            GlobalPool = new BufferPool(messagingOptions.BufferPoolBufferSize, messagingOptions.BufferPoolPreallocationSize);
```

Comment:
Should this be renamed to MinimumBufferSize or something similar? It's no longer the exact size